src/backend/base/langflow/components/tools/google_serper_api.py

```diff
@@ -14,27 +24,87 @@ class GoogleSerperAPIComponent(LCToolComponent):
     inputs = [
         SecretStrInput(name="serper_api_key", display_name="Serper API Key", required=True),
         MultilineInput(
-            name="input_value",
-            display_name="Input",
+            name="query",
+            display_name="Query",
         ),
         IntInput(name="k", display_name="Number of results", value=4, required=True),
+        DropdownInput(
+            name="query_type",
+            display_name="Query Type",
+            required=False,
+            options=["news", "search"],
+            value="search",
+        ),
+        DictInput(
+            name="query_params",
+            display_name="Query Params",
+            required=False,
+            value={
+                "gl": "us",
+                "hl": "en",
+            },
+            list=True,
+        ),
     ]
 
+    class QuerySchema(BaseModel):
+        query: str = Field(..., description="The query to search for.")
+        query_type: str = Field(
+            "search",
+            description="The type of search to perform (e.g., 'news' or 'search').",
+        )
+        k: int = Field(4, description="The number of results to return.")
+        query_params: dict[str, Any] = Field({}, description="Additional query parameters to pass to the API.")
```

Comment:
Please put the class definition outside the Component

Reply:
@ogabrielluiz Done !